pkg/provider/ibm/provider.go

```diff
@@ -294,6 +299,26 @@ func getIamCredentialsSecret(ibm *providerIBM, secretName *string, secretGroupNa
 	return nil, fmt.Errorf("key %s does not exist in secret %s", smAPIKeyConst, *secretName)
 }
 
+func getServiceCredentialsSecret(ibm *providerIBM, secretName *string, secretGroupName string) ([]byte, error) {
+	response, err := getSecretData(ibm, secretName, sm.Secret_SecretType_ServiceCredentials, secretGroupName)
+	if err != nil {
+		return nil, err
+	}
+	secMap, err := formSecretMap(response)
+	if err != nil {
+		return nil, err
+	}
+	if val, ok := secMap[credentialsConst]; ok {
+		var mval []byte
+		mval, err = json.Marshal(val)
+		if err != nil {
+			return nil, err
+		}
+		return mval, nil
```

Comment:
```suggestion
		mval, err := json.Marshal(val)
		if err != nil {
			return nil, fmt.Errorf("failed to marshal sec map for service credentials secret: %w", err)
		}

		return mval, nil
```

Reply:
@Shanti-G Do you agree?

Reply:
Suggested changes have been addressed in a new PR: https://github.com/external-secrets/external-secrets/pull/2967